lib/FusionInventory/Agent/Tools/Win32.pm

```diff
@@ -690,33 +692,60 @@ sub getInterfaces {
             }
         }
 
-        $configurations[$object->{Index}] = $configuration;
+        # XP compatibility
+        $indexKey = 'Index' if !exists $object->{$indexKey};
+
+        $configurations[$object->{$indexKey}] = $configuration;
     }
 
     my @interfaces;
 
-    foreach my $object (getWMIObjects(
-        class      => 'Win32_NetworkAdapter',
-        properties => [ qw/Index PNPDeviceID Speed PhysicalAdapter GUID/ ]
-    )) {
+    # For Win8 or Above
+    my @networkAdapter = getWMIObjects(
+        moniker    => 'winmgmts://./root/StandardCimv2',
+        class      => 'MSFT_NetAdapter',
+        properties => [ qw/InterfaceIndex PnPDeviceID Speed HardwareInterface InterfaceGuid InterfaceDescription/ ]
+    );
+    my $netAdapterKey = {
+        'PNPDeviceID' => 'PnPDeviceID',
+        'GUID' => 'InterfaceGuid',
+        'DESCRIPTION' => 'InterfaceDescription',
+        'PhysicalAdapter' => 'HardwareInterface'
+    };
+    
+    if (!@networkAdapter) {
+        # Legacy for Win<8
+        @networkAdapter = getWMIObjects(
+            class      => 'Win32_NetworkAdapter',
+            properties => [ qw/Index InterfaceIndex PNPDeviceID Speed PhysicalAdapter GUID/ ]
+        );
+        $netAdapterKey = {
+            'PNPDeviceID' => 'PNPDeviceID',
+            'GUID' => 'GUID',
+            'PhysicalAdapter' => 'PhysicalAdapter'
+        };
+    }
+
+
+    foreach my $object (@networkAdapter) {
         # http://comments.gmane.org/gmane.comp.monitoring.fusion-inventory.devel/34
-        next unless $object->{PNPDeviceID};
+        next unless $object->{$netAdapterKey->{PNPDeviceID}};
 
         my $pciid;
-        if ($object->{PNPDeviceID} =~ /PCI\\VEN_(\w{4})&DEV_(\w{4})&SUBSYS_(\w{4})(\w{4})/) {
+        if ($object->{$netAdapterKey->{PNPDeviceID}} =~ /PCI\\VEN_(\w{4})&DEV_(\w{4})&SUBSYS_(\w{4})(\w{4})/) {
             $pciid = join(':', $1 , $2 , $3 , $4);
         }
 
-        my $configuration = $configurations[$object->{Index}];
+        my $configuration = $configurations[$object->{$indexKey}];
```

Comment:
Here, you may want to better try:
```
        my $configuration = $configurations[$object->{InterfaceIndex} || $object->{Index}];
```
But you may also want to re-use `$indexKey` set earlier in L. ~696 in the case `$object->{InterfaceIndex}` could be defined and set to `0` as in that case the `||` would do the wrong choice.
This is always important to check when using this simplified syntax as it will not only work on undefined values but also on false like `''` or `0`.
Another way would be to use another simple private api like:
```
sub _getObjectIndex {
    my ($object) = @_;
    if (defined($object->{InterfaceIndex})) {
        return $object->{InterfaceIndex};
    } else {
        return $object->{Index};
    }
}
```
And even better would be to use a `FusionInventory::Agent::Tool::Win32::Network` class...